crates/uv-scripts/src/lib.rs

```diff
@@ -43,28 +67,16 @@ pub struct Pep723Metadata {
     pub dependencies: Option<Vec<pep508_rs::Requirement<VerbatimParsedUrl>>>,
     pub requires_python: Option<pep440_rs::VersionSpecifiers>,
     pub tool: Option<Tool>,
+    /// The raw unserialized document.
+    #[serde(skip)]
+    pub raw: String,
 }
 
 impl Pep723Metadata {
-    /// Read the PEP 723 `script` metadata from a Python file, if it exists.
-    ///
-    /// See: <https://peps.python.org/pep-0723/>
-    pub async fn read(file: impl AsRef<Path>) -> Result<Option<Self>, Pep723Error> {
-        let contents = match fs_err::tokio::read(file).await {
-            Ok(contents) => contents,
-            Err(err) if err.kind() == io::ErrorKind::NotFound => return Ok(None),
-            Err(err) => return Err(err.into()),
-        };
-
-        // Extract the `script` tag.
-        let Some(contents) = extract_script_tag(&contents)? else {
-            return Ok(None);
-        };
-
-        // Parse the metadata.
-        let metadata = toml::from_str(&contents)?;
-
-        Ok(Some(metadata))
+    /// Parse `Pep723Metadata` from a raw TOML string.
+    pub fn from_string(raw: String) -> Result<Self, toml::de::Error> {
```

Comment:
Nit: this should be `from_str`.

Reply:
Clippy is not happy with `from_str`

```
warning: method `from_str` can be confused for the standard trait method `std::str::FromStr::from_str`
```j

Reply:
I think it's ok to implement the trait instead of as an associated method here. But I can also do it before merging and show you as an example, either is fine.

Reply:
Done